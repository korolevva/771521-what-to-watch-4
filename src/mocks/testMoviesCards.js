export const moviesCards = [
  {
    id: 1,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    imagePreview: `img/bohemian-rhapsody.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Bohemian Rhapsody`,
    genre: `Biography`,
    date: 2018,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury`,
    rating: 8.0,
    ratingCount: 305,
    director: `Bryan Singer`,
    stars: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    duration: 167,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    background: `https://placeimg.com/1300/552/tech`,
    imagePreview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    date: 2018,
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    rating: 6.6,
    ratingCount: 423,
    director: `David Yates`,
    stars: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    duration: 90,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];
