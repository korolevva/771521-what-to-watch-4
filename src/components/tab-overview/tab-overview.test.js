import React from "react";
import renderer from "react-test-renderer";
import TabOverview from "./tab-overview.jsx";
import {moviesCards} from "../../mocks/testMoviesCards.js";

const card = moviesCards[0];

it(`Render TabOverview`, () => {
  const tree = renderer
    .create(<TabOverview
      card={card}
    />,
    {
      createNodeMock: () => {
        return {};
      }
    }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
