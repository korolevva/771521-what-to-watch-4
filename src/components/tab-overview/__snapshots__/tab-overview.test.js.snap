// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TabOverview 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        305
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      The story of the legendary British rock band Queen and lead singer Freddie Mercury
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Bryan Singer
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Rami Malek, Lucy Boynton, Gwilym Lee
      </strong>
    </p>
  </div>,
]
`;
